File: 1_encapsulation/2_cpp/1_define.cc
#include <iostream>
#include <string>

class Sensor
{
public:
    Sensor(const std::string &id, const std::string &loc) : sensor_id(id), location(loc) {}
    std::string sensor_id;
    std::string location;
};

int main()
{
    Sensor temp_sensor("T1", "Living Room");
    std::cout << "Sensor ID: " << temp_sensor.sensor_id << ", Location: " << temp_sensor.location << std::endl;
    return 0;
}

File: 1_encapsulation/1_python/1_define.py
class Sensor:
    def __init__(self, sensor_id, location):
        self.sensor_id = sensor_id
        self.location = location

# 创建对象
temp_sensor = Sensor(sensor_id="T1", location="Living Room")
print(f"Sensor ID: {temp_sensor.sensor_id}, Location: {temp_sensor.location}")
